x-pack/legacy/plugins/ingest/server/libs/adapters/policy/default.ts

```diff
@@ -39,192 +47,133 @@ export class PolicyAdapter {
       ...policySO.attributes,
     };
 
-    const decoded = RuntimePolicyFile.decode(policy);
+    const decoded = RuntimeStoredPolicy.decode(policy);
 
     if (isRight(decoded)) {
       return decoded.right;
     } else {
       throw new Error(
-        `Invalid PolicyFile data. == ${JSON.stringify(policy)} -- ${PathReporter.report(decoded)}`
+        `Invalid Policy data. == ${JSON.stringify(policy)} -- ${PathReporter.report(decoded)}`
       );
     }
   }
 
   public async list(
-    page: number = 1,
-    perPage: number = 25
-  ): Promise<{ items: PolicyFile[]; total: number }> {
-    const policys = await this.so.find<any>({
+    user: FrameworkUser,
+    options: ListOptions = {}
+  ): Promise<{ items: StoredPolicy[]; total: number; page: number; perPage: number }> {
+    const { page = 1, perPage = 20, kuery } = options;
+    const filters = [];
+
+    if (kuery && kuery !== '') {
+      filters.push(kuery.replace(/policies\./g, 'policies.attributes.'));
```

Comment:
could you explain why this is needed? (a comment in the code would be great!)

Reply:
It's so users using KQL won't need to know SO data formatting